k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.ipp

```diff
@@ -174,6 +176,29 @@ std::vector<CollNamePair> convertReconstructedParticles(const std::string& name,
   return results;
 }
 
+// Go from chi^2 and probability (1 - CDF(chi^2, ndf))
+// to ndf by a binary search
+int find_ndf(double chi2, double prob) {
+  int upper = 100;
+  while (TMath::Prob(chi2, upper) < prob) {
+    upper *= 2;
+  }
```

Comment:
Just for my understanding: This part is not technically part of the binary search, but rather here to determine an upper bound for that, that is then used in the actual binary search?

Reply:
Indeed, if the initial guess of the upper bound is below `ndf` then the actual value will never be found since this always looks between the lower and upper bounds.

Reply:
Makes sense. Maybe put a comment here why we start at 100?